vector/src/main/java/im/vector/fragments/VectorSettingsNotificationsTroubleshootFragment.kt

```diff
@@ -0,0 +1,160 @@
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.TextView
+import im.vector.R
+import im.vector.activity.MXCActionBarActivity
+import im.vector.adapters.TroubleshootTest
+import im.vector.push.fcm.NotificationTroubleshootTestManager
+import im.vector.util.BugReporter
+
+/*
+ * Copyright 2018 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+class VectorSettingsNotificationsTroubleshootFragment : VectorBaseFragment() {
+
+    lateinit var mRecyclerView: RecyclerView
+    lateinit var mBottomView: View
+    lateinit var mSummaryTitle: TextView
+    lateinit var mSummaryDescription: TextView
+    lateinit var mSummaryButton: Button
+    lateinit var mRunButton: Button
+
+    var testManager: NotificationTroubleshootTestManager? = null
+
+    override fun getLayoutResId() = R.layout.fragment_settings_notifications_troubleshoot
+
+
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        mRecyclerView = view.findViewById(R.id.troubleshoot_test_recycler_view);
+        val layoutManager = LinearLayoutManager(context)
+        mRecyclerView.layoutManager = layoutManager
+
+        val dividerItemDecoration = DividerItemDecoration(mRecyclerView.context,
+                layoutManager.orientation);
+        mRecyclerView.addItemDecoration(dividerItemDecoration)
+
+        mBottomView = view.findViewById(R.id.bottomView)
+        mSummaryTitle = view.findViewById(R.id.summ_title)
+        mSummaryDescription = view.findViewById(R.id.summ_description)
+        mSummaryButton = view.findViewById(R.id.summ_button)
+        mRunButton = view.findViewById(R.id.runButton)
+
+        mSummaryButton.setOnClickListener {
+            BugReporter.sendBugReport()
+        }
+
+        mRunButton.setOnClickListener() {
+            testManager?.retry()
+        }
+
+        startUI()
+    }
+
+    private fun startUI() {
+
+        mSummaryTitle.text = getString(R.string.settings_troubleshoot_diagnostic)
+        mSummaryDescription.text = getString(R.string.settings_troubleshoot_diagnostic_running_status)
+        mSummaryButton.text = getText(R.string.send_bug_report)
+
+        testManager = NotificationTroubleshootTestManager(this)
+
+        testManager?.statusListener = {
+            if (isAdded) {
+                (mBottomView as? ViewGroup)?.let {
```

Comment:
Why this cast check?

Reply:
changed type of mBottomView to ViewGroup